framework/source/class/qx/ui/layout/HBox.js

```diff
@@ -371,6 +383,12 @@ qx.Class.define("qx.ui.layout.HBox",
               max : hint.maxWidth,
               flex : flex
             };
+            if (notEnoughSpace) {
+              var props = children[i].getLayoutProperties();
+              if (props && props.flexShrink) {
+                flexibles[i].min = 0;
```

Comment:
Very, very cooll feature! It looks, though, like this eliminates the minWidth of all widgets in the layout, when there's not enough space, so the user has no control. What do you think about flexShrink being something other than a boolean, i.e., giving it a way to specify which child or children should be considered for further shrinking beyond their minWidth?

Reply:
`flexShrink` is set on the child's layout properties, so it only applies to that one child :)